DemoApp/Sources/Views/WaitingLocalUserView/DemoWaitingLocalUserView.swift

```diff
@@ -30,7 +31,7 @@ struct DemoWaitingLocalUserView<Factory: DemoAppViewFactory>: View {
             viewFactory
                 .makeInnerWaitingLocalUserView(viewModel: viewModel)
 
-            if chatViewModel?.isChatVisible == true {
```

Comment:
hmm, I don't think you need additional State for this - you already have the same one in place in the view model.
I notice that the check is different now (`== false` instead of `== true`). Could that be the fix instead?

Reply:
The injected chatViewModel isn't posting updates, which means that this won't change when the chat presents/hides. 

I think previously we had a mistake and lately the popup isn't visible when the chat is closed.

Reply:
why not make it `@ObservedObject`?
```swift
@ObservedObject var chatViewModel = InjectedValues[\.chatViewModel]!
```

Reply:
Want to avoid explicitely unwrapping. Do you see any issues with the private @State?